### PR TITLE
Client should not assume launch engine has completed on exception

### PR DESCRIPTION
**Pull request by pan3793**
### _Why are the changes needed?_

`LaunchEngineOp` will never close until session is closed, the client should not assume launch engine has completed on exception.

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] [Run test](https://kyuubi.apache.org/docs/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

**Comment by pan3793** [comment]
cc @gabrywu

**Review by turboFei — APPROVED** [review]

**Comment by yaooqinn** [comment]
shall we test this?

**Review by gabrywu — APPROVED** [review]
+1

**Comment by pan3793** [comment]
> shall we test this?

Actually, we have a test for this case, and I enhance it in this PR

https://github.com/apache/incubator-kyuubi/blob/8de2f5f17fa909979294a879503700bd3b083df0/kyuubi-server/src/test/scala/org/apache/kyuubi/operation/KyuubiOperationPerConnectionSuite.scala#L116-L130

File: kyuubi-common/src/main/scala/org/apache/kyuubi/operation/Operation.scala [hunk 1]
```diff
@@ -49,4 +49,7 @@ trait Operation {
 object Operation {
   val DEFAULT_FETCH_ORIENTATION_SET: Set[FetchOrientation] =
     Set(FetchOrientation.FETCH_NEXT, FetchOrientation.FETCH_FIRST, FetchOrientation.FETCH_PRIOR)
+
+  val LAUNCH_ENGINE_GUID = "kyuubi.session.engine.launch.handle.guid"
```
Comment (yaooqinn) [inline kyuubi-common/src/main/scala/org/apache/kyuubi/operation/Operation.scala:53]: what about add them to KyuubiReservedKeys

**Review by yaooqinn — APPROVED** [review]

**Pull request closed without merging**